tests/lib/e2e/buckets/files.e2e-spec.ts

```
@@ -0,0 +1,400 @@
+        expect(response.status).toBe(200);
+        const body = response.body;
+
+        expect(body.bucket).toBe(bucketId)
+        expect(body.created).toBeDefined()
+        expect(body.index).toBe(index)
+        expect(body.shards).toHaveLength(2)
+
+        const [firstShard, secondShard] = body.shards
+
+        expect(firstShard.hash).toBeDefined()
+        expect(firstShard.url).toBeDefined()
+        expect(secondShard.hash).toBeDefined()
+        expect(secondShard.url).toBeDefined()
+
+      })
+
+    })
+
+  })
+
+  describe('File Management v1', () => {
+    describe('Deleting a file', () => {
+
+      it('When a user wants to delete a file, it should work if the file and the bucket exist', async () => {
+        // Arrange: Create a bucket
+        const { body: { id: bucketId } } = await testServer
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+
+        // Arrange: start the upload
+        const { body: { uploads } } = await testServer.post(`/v2/buckets/${bucketId}/files/start`)
+          .set('Authorization', getAuth(testUser))
+          .send({ uploads: [{ index: 0, size: 1000, }, { index: 1, size: 10000, },], })
+
+        // Arrange: finish the upload
+        const index = crypto.randomBytes(32).toString('hex');
+        const { body: file } = await testServer.post(`/v2/buckets/${bucketId}/files/finish`)
+          .set('Authorization', getAuth(testUser))
+          .send({
+            index,
+            shards: (uploads as any[]).map(upload => ({ hash: crypto.randomBytes(20).toString('hex'), uuid: upload.uuid, })),
+          });
+
+
+        // Act: remove the file
+        const response = await testServer.delete(`/buckets/${bucketId}/files/${file.id}`)
+          .set('Authorization', getAuth(testUser))
+
+        // Assert
+        expect(response.status).toBe(204);
+
+      })
+
+      it('When a user wants to delete a file, it should work if the file does not exist', async () => {
```

'It should fail/It should return an error', isn't it?

---

Yes